datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java

```diff
@@ -388,6 +388,10 @@ protected void trackEventsProcessedProgress(int recordCount) {
     }
   }
 
+  protected ConsumerRecords<?, ?> consumerPoll(long pollInterval) {
+    return _consumer.poll(pollInterval);
```

Comment:
poll(long) has been deprecated in favor of poll(Duration) per KIP-266 -- maybe time to make the switch?

Reply:
For some reasons, Kafka behaves differently when poll(Duration.Zero) and poll((long)0). It looks like I cannot switch to new api without breaking the compatibility.